doc/man3/SSL_new_listener.pod

```diff
@@ -0,0 +1,209 @@
+support other protocols, such as TLS over TCP, plain TCP or DTLS in future.
+
+=head1 CLIENT-ONLY USAGE
+
+It is also possible to use the listener interface without accepting any
+connections and without listening for connections. This can be useful in
+circumstances where it is desirable for multiple connections to share the same
+underlying network resources. For example, multiple outgoing QUIC client
+connections could be made to use the same underlying UDP socket.
+
+To use client-only mode, pass the flag B<SSL_LISTENER_FLAG_NO_ACCEPT> when
+calling SSL_new_listener(). In this mode, SSL_listen() still begins the process
+of handling network resources, but incoming connections are never accepted.
+Calling SSL_accept_connection() is an error and will return NULL. One or more
+outgoing connections under a listener can then be created using the call
+SSL_new_from_listener().
+
+The SSL_new_from_listener() creates a client connection under a given listener
+SSL object. For QUIC, it is also possible to use SSL_new_from_listener() in
+conjunction with a listener which does accept incoming connections (i.e., which
+was not created using B<SSL_LISTENER_FLAG_NO_ACCEPT>), leading to a UDP network
+endpoint which has both incoming and outgoing connections.
+
+The I<flags> argument of SSL_new_from_listener() is reserved and must be set to
+0.
+
+Creating a listener using a B<SSL_CTX> which uses a client-oriented
+B<SSL_METHOD> such as L<OSSL_QUIC_client_method(3)> or
+L<OSSL_QUIC_client_thread_method(3)> automatically implies the
+B<SSL_LISTENER_FLAG_NO_ACCEPT> flag. The B<SSL_LISTENER_FLAG_NO_ACCEPT> flag may
+optionally also be specified in this case but is ignored. This is an alternative
+way of using the listener functionality in client-only mode.
+
+=head1 RETURN VALUES
+
+SSL_new_listener() returns a new listener SSL object or NULL on failure.
+
+SSL_is_listener() returns 0 or 1 depending on the type of the SSL object on
+which it is called.
+
+SSL_get0_listener() returns an SSL object pointer (potentially to the same
+object on which it is called) or NULL.
+
+SSL_listen() returns 1 on success or 0 on failure.
+
+SSL_accept_connection() returns a pointer to a new SSL object on success or NULL
+on failure. On success, the caller assumes ownership of the reference.
+
+SSL_get_accept_connection_queue_len() returns a nonnegative value, or 0 if
+called on an unsupported SSL object type.
+
+SSL_new_from_listener() returns a pointer to a new SSL object on success or NULL
+on failure. On success, the caller assumes ownership of the reference.
+
+=head1 SEE ALSO
+
+L<OSSL_QUIC_server_method(3)>, L<SSL_free(3)>, L<SSL_set_bio(3)>,
+L<SSL_handle_events(3)>, L<SSL_get_rpoll_descriptor(3)>,
+L<SSL_set_blocking_mode(3)>
+
+=head1 HISTORY
+
+OSSL_QUIC_client_method() and OSSL_QUIC_client_thread_method() were added in
+OpenSSL 3.2.
+
+OSSL_QUIC_server_method() was added in OpenSSL @VERSION_QUIC_SERVER@.
```

Comment:
This HISTORY section looks like a copy&paste from somewhere else. Needs updating for the functions described on the this page.